File: Programmers/jaehyung/짝지어제거하기.java
package Programmers.jaehyung;

/**
 * ㅁㅁㅁㅁ [골드 4] (미성공)
 * https://www.acmicpc.net/problem/0000
 */

import java.io.IOException;
import java.util.*;

public class 짝지어제거하기 {

  public int solution(String s) {
    Stack<Character> stack = new Stack();
    for (char c : s.toCharArray()){
      if(!stack.isEmpty() && stack.peek() == c){
        stack.pop();
      } else {
        stack.push(c);
      }
    }
    return stack.isEmpty()? 1: 0;
  }

  public static void main(String[] args) throws IOException {
    짝지어제거하기 T = new 짝지어제거하기();
    Input[] inputList = new Input[]{
            new Input("baabaa", 1),
            new Input("cdcd", 0)
    };
    for (Input input : inputList) {
      int answer = T.solution(input.s);
      System.out.println("입력값: " + input.s);
      System.out.println("답: " + input.answer);
      System.out.println("출력값: " + answer);
      System.out.println(answer == input.answer? "테스트 성공" : "테스트 실패");
      System.out.println();
    }
  }

  static class Input{
    // input
    String s;

    // 정답
    int answer;

    public Input(String s, int answer){
      this.s = s;
      this.answer = answer;
    }
  }
}

File: Programmers/seongyong/짝지어 제거하기.py
def solution(s):
    stack = []
    for i in range(len(s)):
        if not stack:
            stack.append(s[i])          # stack이 비어있다면 push()
        else:
            if s[i] == stack[-1]:       # stack 마지막 값과 s[i]가 같다면 pop()
                stack.pop()
            else:
                stack.append(s[i])      # stack 마지막 값과 s[i]가 다르면 push()

    if stack : return 0                 # stack이 비어있지 않다면 return 0
    else : return 1
